openmetadata-airflow-apis/openmetadata_managed_apis/workflows/ingestion/common.py

```diff
@@ -115,25 +127,57 @@ def build_source(ingestion_pipeline: IngestionPipeline) -> WorkflowSource:
             sourceConfig=ingestion_pipeline.sourceConfig,
         )
 
-    if service_type == "databaseService":
-        service: DatabaseService = metadata.get_by_name(
-            entity=DatabaseService, fqn=ingestion_pipeline.service.name
-        )
-    elif service_type == "pipelineService":
-        service: PipelineService = metadata.get_by_name(
-            entity=PipelineService, fqn=ingestion_pipeline.service.name
-        )
-    elif service_type == "dashboardService":
-        service: DashboardService = metadata.get_by_name(
-            entity=DashboardService, fqn=ingestion_pipeline.service.name
-        )
-    elif service_type == "messagingService":
-        service: MessagingService = metadata.get_by_name(
-            entity=MessagingService, fqn=ingestion_pipeline.service.name
-        )
-    elif service_type == "mlmodelService":
-        service: MlModelService = metadata.get_by_name(
-            entity=MlModelService, fqn=ingestion_pipeline.service.name
+    entity_class = None
+    try:
+        if service_type == "databaseService":
+            entity_class = DatabaseService
+            service: DatabaseService = metadata.get_by_name(
+                entity=DatabaseService, fqn=ingestion_pipeline.service.name
+            )
```

Comment:
we can use the `entity_class` for the `entity` argument.

Reply:
Hi @TeddyCr @ulixius9 sorry for jumping in. I am not sure I understand the suggestion or the proposed fix 🥲 

Could you please elaborate? The current commit seems more verbose than it has to be.

Thanks!

Reply:
There was an `entity_class` added to fetch the suffix for the entity class and provide a better error message.

Reply:
aaaah got it got it missed the exception later on. Thanks!